test/integration/binary/launchers.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "launchers.hpp"
+
+#include <gtest/gtest.h>
+#include "common/byteutils.hpp"
+
+#include "bindings/model_crypto.hpp"
+
+using namespace boost::process;
+
+namespace binary_test {
+
+  constexpr auto cTimeToKill = std::chrono::minutes(15);
+
+  void Launcher::operator()(const std::string &example) {
+    ipstream pipe;
+    const auto &command = launchCommand(example);
+    if (command.empty()) {
+      FAIL() << "Launcher provided empty command";
+    }
+    child c(command, std_out > pipe);
+    auto terminated = c.wait_for(cTimeToKill);
+    if (not terminated) {
+      c.terminate();
+      FAIL() << "Child process was terminated because execution time limit "
+                "has been exceeded";
+    }
+    readBinaries(pipe);
+  }
+
+  void Launcher::readBinaries(ipstream &stream) {
+    transactions.clear();
+    queries.clear();
+    std::string line;
+    bool first_line(true);
+    while (stream and std::getline(stream, line) and not line.empty()) {
+      iroha::protocol::Transaction proto_tx;
+      iroha::protocol::Query proto_query;
+      if (auto byte_string = iroha::hexstringToBytestring(line)) {
```

Comment:
monadic bind?

Reply:
It is possible to use monadic bind here but it is just an alternative way of implementation. Let's keep current version because suggested change does not bring visible benefits.

Reply:
Code become cleaner since you don't need to get explicitly get value from optional. Also at the codebase it's the preferred style